File: android/app/src/main/java/com/lightningkite/khrysalistemplate/vg/BleClientDemoVG.shared.kt
package com.lightningkite.khrysalistemplate.vg

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.*
import com.lightningkite.khrysalis.bluetooth.Ble
import com.lightningkite.khrysalis.bluetooth.PropertyBleCharacteristicServer
import com.lightningkite.khrysalis.lifecycle.and
import com.lightningkite.khrysalis.lifecycle.appInForeground
import com.lightningkite.khrysalis.lifecycle.closeWhenOff
import com.lightningkite.khrysalis.lifecycle.lifecycle
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalis.observables.*
import com.lightningkite.khrysalis.observables.binding.*
import com.lightningkite.khrysalistemplate.R
import com.lightningkite.khrysalistemplate.layouts.*
import java.util.*

@Suppress("NAME_SHADOWING")
class BleClientDemoVG(val deviceId: String) : ViewGenerator() {

    override val title: String get() = "BleHostDemo"
    override fun generate(dependency: ViewDependency): View {
        val xml = BleClientDemoXml()
        val view = xml.setup(dependency)

        //OpenConnectionObservable -> Event -> Observable
        //Is a DRF just an event?!
        //Ble.connect(dependency, deviceId).flatMap { it?.subscribe() }.asObservable(default = "")

//        xml.status.bindString(server.clients.map { it -> "Serving ${it.size} clients" })
//        xml.charValue.bindString(characteristic.map(
//            read = { it -> it.toString(Charsets.UTF_8) },
//            write = { it -> it.toByteArray(Charsets.UTF_8) }
//        ))

        return view
    }
}

File: ios/Khrysalis Template/com/lightningkite/khrysalistemplate/vg/BleClientDemoVG.shared.swift
//Package: com.lightningkite.khrysalistemplate.vg
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay
import KhrysalisMaps



public class BleClientDemoVG: ViewGenerator {
    
    public var deviceId: String
    
    override public var title: String {
        get {
            return "BleHostDemo"
        }
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = BleClientDemoXml()
        var view = xml.setup(dependency)
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    public init(deviceId: String) {
        self.deviceId = deviceId
        super.init()
    }
    convenience public init(_ deviceId: String) {
        self.init(deviceId: deviceId)
    }
}
 
